src/server/src/main/java/io/cassandrareaper/core/Stream.java

```diff
@@ -0,0 +1,289 @@
+    this.lastUpdated = lastUpdated;
+  }
+
+  public void setCompleted(boolean completed) {
+    this.completed = completed;
+  }
+
+  public void setSuccess(boolean success) {
+    this.success = success;
+  }
+
+  public String streamId() {
+    return streamId(this.host, this.peer);
+  }
+
+  public static String streamId(String host, String peer) {
+    return String.format("%s->%s", host, peer.replaceAll("/", ""));
+  }
+
+  public String toString() {
+    return String.format(
+        "Stream(host=%s, peer=%s, toSend=%d, sent=%d, toReceive=%d, received=%d, updated=%d, completed=%b",
+        host, peer, sizeToSend, getSizeSent(), sizeToReceive, getSizeReceived(), lastUpdated, completed);
+  }
+
+  public static Stream.Builder builder() {
+    return new Stream.Builder();
+  }
+
+  @JsonPOJOBuilder(buildMethodName = "build", withPrefix = "with")
+  public static final class Builder {
+    private String host;
+    private String peer;
+    private long sizeToSend;
+    private long sizeToReceive;
+    private Map<String, FileProgress> progressSent;
+    private Map<String, FileProgress> progressReceived;
+    private long lastUpdated;
+    private boolean completed;
+    private boolean success;
+
+    private Builder() {}
+
+    public Builder withHost(String host) {
+      this.host = host;
+      return this;
+    }
+
+    public Stream.Builder withPeer(String peer) {
+      this.peer = peer;
+      return this;
+    }
+
+    public Stream.Builder withProgressReceived(Map<String, FileProgress> progressReceived) {
+      this.progressReceived = progressReceived;
+      return this;
+    }
+
+    public Stream.Builder withSizeToReceive(long sizeToReceive) {
+      this.sizeToReceive = sizeToReceive;
+      return this;
+    }
+
+    public Stream.Builder withProgressSent(Map<String, FileProgress> progressSent) {
+      this.progressSent = progressSent;
+      return this;
+    }
+
+    public Stream.Builder withSizeToSend(long sizeToSend) {
+      this.sizeToSend = sizeToSend;
+      return this;
+    }
+
+    public Stream.Builder withLastUpdated(long lastUpdated) {
+      this.lastUpdated = lastUpdated;
+      return this;
+    }
+
+    public Stream.Builder withCompleted(boolean completed) {
+      this.completed = completed;
+      return this;
+    }
+
+    public Stream.Builder withSuccess(boolean success) {
+      this.success = success;
+      return this;
+    }
+
+    public Stream build() {
+      return new Stream(this);
+    }
+
+  }
+
+  public static class FileProgress {
+    public long totalBytes;
+    public long currentBytes;
+
+    public FileProgress(long totalBytes, long currentBytes) {
+      this.totalBytes = totalBytes;
+      this.currentBytes = currentBytes;
+    }
+  }
+
+  private static FileProgress toFileProgress(ProgressInfo pi) {
+    return new FileProgress(pi.totalBytes, pi.currentBytes);
+  }
+
+  public static Stream newStream(String host, SessionInfo sessionInfo, long timeStamp) {
```

Comment:
`newStream` should not be part of this class file as it's a factory method to create Stream objects. The Stream POJO should only contain getters/setters and the builder class.
You could create a StreamFactory class that would contain this method (and the others we could need in the future).